devdoc/doublylinkedlist_requirements.md

```diff
@@ -104,3 +108,17 @@ extern PDLIST_ENTRY DList_RemoveHeadList(PDLIST_ENTRY listHead);
 **SRS_DLIST_06_013: [** DList_RemoveHeadList shall return listHead if that's the only item in the list. **]**
 
 Note: The Flink & Blink of the returned PDLIST_ENTRY shall be undefined.
+
+### DList_ForEach
+```c
+MOCKABLE_FUNCTION(, int, DList_ForEach, PDLIST_ENTRY, listHead, DLIST_ACTION_FUNCTION, actionFunction, void*, actionContext);
```

Comment:
>listHead [](http://example.com/codeflow?start=54&length=8)

This should also be checked for NULL.
Frankly like I mentioned in my other comment all other functions should be consistent with how we write code and shall check their arguments.

Reply:
The new code at least should have the checks since we're writing it now.
You can create a separate task for the existing functions to check their args.

Reply:
Since this comment was already ignored once, I'll still wait for it until it's done :-)

Reply:
Sorry, didn't mean to ignore the comment. I was just trying to follow the precedent in the module. 

Added arg checks for new code. 

Added task in backlog to add checks everywhere: [https://msazure.visualstudio.com/DefaultCollection/One/_backlogs/backlog/Azure%20Messaging%20Store/Backlog%20items/?workitem=12862713](https://msazure.visualstudio.com/DefaultCollection/One/_backlogs/backlog/Azure%20Messaging%20Store/Backlog%20items/?workitem=12862713)